src/unittest_parametrize/__init__.py

```diff
@@ -48,7 +51,13 @@ def test(
                     _params: dict[str, Any] = params,
                     **kwargs: Any,
                 ) -> Any:
-                    return _func(self, *args, **_params, **kwargs)
+                    try:
+                        return _func(self, *args, **_params, **kwargs)
+                    except AssertionError as e:
+                        # Suppress warning about not closing over test_name
+                        tn = test_name  # noqa: B023
```

Comment:
Just a comment on this, the B023 suppression in this instance because the test_name is going to be the same for each returned func, and I didn't want to make it even more convoluted with the workarounds of a default arg or using partial.